src/main/java/core/basesyntax/service/OperationStrategyImpl.java

```
@@ -0,0 +1,18 @@
+package core.basesyntax.service;
+
+import java.util.Map;
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.service.operation.OperationHandler;
+
+public class OperationStrategyImpl implements OperationStrategy {
+    private final Map<FruitTransaction.Operation, OperationHandler> operationHandlers;
+
+    public OperationStrategyImpl(Map<FruitTransaction.Operation, OperationHandler> operationHandlers) {
+        this.operationHandlers = operationHandlers;
+    }
+
+    @Override
+    public OperationHandler get(FruitTransaction.Operation operation) {
+        return operationHandlers.get(operation);
```

This method returns `null` when there is no handler mapped for the provided operation. The task requires handling incorrect/unknown operation codes or incorrect strategy mapping with an informative exception rather than silently skipping. Quote: "Incorrect/unknown operation code or incorrect strategy mapping must be handled (e.g., throw informative RuntimeException)" . Change this method to throw a RuntimeException (with the missing operation in the message) when a handler is not found.